docs/proposals/20200804-windows-support.md

```diff
@@ -0,0 +1,271 @@
+
+#### Signing of the components.  
+Some infrastructure providers will require any scripts and binaries are signed before deployment.  
+This will be managed by providing the ability to provide external scripts and binaries during the VHD building process.  We will also be discussing this further with the Image Builder repository team during the process for adding the VHDs.
+
+#### Known prototypes and prior work: 
+- https://github.com/adelina-t/cloudbase-init-capz-demo
+- https://github.com/benmoss/kubeadm-windows/tree/master/cluster-api-aws
+
+### Security Model
+
+Wins is currently [recommended way to use kubeadm](https://kubernetes.io/docs/tasks/administer-cluster/kubeadm/adding-windows-nodes/).  
+Limiting access to the named pipes that are required is one way to mitigate access.   Wins is currently 
+required during and after provisioning for running kube-proxy and the CNI daemonset.  
+
+The security model for Privileged containers is still being discussed in KEP and is still early and will be
+addressed by CAPI when the KEP progresses.
+
+Kubeadm bootstrap token should be able to use multi-part mime documents for cloudbase-init as done for [Linux in CAPA](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/28d01d064cc2e5b0286ae23b3be7203f18b00447/controllers/awsmachine_controller.go#L601). 
+This will require an update to Cloudbase-init which does support [mutli-part mime documents](https://cloudbase-init.readthedocs.io/en/latest/userdata.html#multi-part-content) but is missing the [boothook](https://cloudinit.readthedocs.io/en/latest/topics/format.html?highlight=boothook#cloud-boothook) functionality.  
+Support for cloudhooks can be added to cloudbase-init to meet the AWS provider requirement that 
+would bring parity to Linux implementation during the Beta phase.
+
+There is [no known requirement](https://github.com/kubernetes-sigs/cluster-api/issues/2218) for managing the Admin
+Kubeconfig or domain passwords in the Windows configuration. Domain passwords should be managed outside the scope of 
+CAPI and only Kubeadm bootstrap tokens, which have limited lifetime, should be used for the joining the Windows nodes. 
+
+### Risks and Mitigations
+
+- Privileged containers are not implemented.
+  - There is an active discussion and [KEP](https://docs.google.com/document/d/12EUtMdWFxhTCfFrqhlBGWV70MkZZPOgxw0X-LTR0VAo/edit#) In place.  At the Beta stage the community can do a checkpoint to determine if the solution fits customer needs
+- Cloudbase-init is a third party dependency
+  - This project is under Apache 2.0 License : https://github.com/cloudbase/cloudbase-init which is cleared under the CNCF Allow list: https://github.com/cncf/foundation/blob/master/allowed-third-party-license-policy.md
+- Windows VHD Distribution
+  - Cluster API already provides the ability to use customer provided VHD and images provided are only for testing demonstration purposes and it is recommended the customer creates their own image. 
```

Comment:
This is actually infrastructure provider specific

Reply:
updated to reflect that it is the infra providers that can provide this ability